registration/tests/models.py

```diff
@@ -471,17 +474,324 @@ def test_resend_activation_email_expired_user(self):
+
+        self.assertEqual(len(mail.outbox[0].alternatives), 0)
+
+    def test_valid_admin_approval(self):
+        """
+        Approving an already activated user's account makes the user
+        active
+        """
+
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+        activated = (SupervisedRegistrationProfile.objects
+                     .activate_user(profile.activation_key))
+
+        self.failIf(isinstance(activated, UserModel()))
+        self.assertEqual(activated, True)
+
+        user = self.registration_profile.objects.admin_approve_user(
+            profile.id, Site.objects.get_current())
+        self.failUnless(isinstance(user, UserModel()))
+        self.assertEqual(user.is_active, True)
+
+    def test_admin_approval_not_activated(self):
+        """
+        Approving a non activated user's account fails
+        """
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+
+        user = self.registration_profile.objects.admin_approve_user(
+            profile.id, Site.objects.get_current())
+        self.failIf(isinstance(user, UserModel()))
+        self.assertEqual(user, False)
+        self.assertEqual(profile.user.is_active, False)
+
+    def test_admin_approval_already_approved(self):
+        """
+        Approving an already approved user's account returns the User model
+        """
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+        activated = (SupervisedRegistrationProfile.objects
+                     .activate_user(profile.activation_key))
+
+        self.failIf(isinstance(activated, UserModel()))
+        self.assertEqual(activated, True)
+
+        user = self.registration_profile.objects.admin_approve_user(
+            profile.id, Site.objects.get_current())
+        self.failUnless(isinstance(user, UserModel()))
+        self.assertEqual(user.is_active, True)
+
+    def test_admin_approval_nonexistent_id(self):
+        """
+        Approving a non existent user profile does nothing and returns False
+        """
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+
+        user = self.registration_profile.objects.admin_approve_user(
+            profile.id, Site.objects.get_current())
+        self.failIf(isinstance(user, UserModel()))
+        self.assertEqual(user, False)
+
+    def test_activation_already_activated(self):
```

Comment:
This is up for discussion. 
``RegistrationProfile.activate_user()`` includes the following:


    if profile.activated:
        # The User has already activated and is trying to activate
        # again. If the User is active, return the User. Else,
        # return False as the User has been deactivated by a site
        # administrator.
            if profile.user.is_active:
                return profile.user
            else:
                return False

When using the ``default`` backend, if a user tries to activate again **and** ``is_active`` is ``True`` this the user instance as a sign. This logic indicates that the user has activated his account succesfully. 

However, when using the ``admin_approval`` backend, if the user has completed activation succesfully ``is_active`` is ``False``. So this test will fail if it is inherited.

The issue is that when using different backends and by following the same procedure ``RegistrationProfile.activate_user()`` returns different results. 
I am not sure if you consider this an issue. Think about it and let me know. I think that this test, however needs to stay.